src/Adapter/Order/OrderDetailUpdater.php

```diff
@@ -199,15 +199,15 @@ public function updateOrderDetailsTaxes(Order $order): void
                             'total_amount' => (float) $totalAmount,
                         ];
                         $orderDetail->unit_price_tax_incl = (float) $orderDetail->unit_price_tax_excl + $unitAmount;
-                        $orderDetail->total_price_tax_incl = (float) $orderDetail->total_price_tax_incl + $totalAmount;
+                        $orderDetail->total_price_tax_incl = (float) $orderDetail->total_price_tax_excl + $totalAmount;
                     }
 
                     Db::getInstance()->insert('order_detail_tax', $orderDetailTaxes, false);
                 }
 
                 // Update OrderDetail values
                 $orderDetail->unit_price_tax_incl = (float) $orderDetail->unit_price_tax_excl + $unitAmount;
-                $orderDetail->total_price_tax_incl = (float) $orderDetail->total_price_tax_incl + $totalAmount;
+                $orderDetail->total_price_tax_incl = (float) $orderDetail->total_price_tax_excl + $totalAmount;
```

Comment:
@kpodemski I think it makes sense `total_price_tax_incl` is `total_price_tax_excl` + `$totalAmount` of taxes.

However, why there are the same lines twice?

```
$orderDetail->unit_price_tax_incl = (float) $orderDetail->unit_price_tax_excl + $unitAmount;
$orderDetail->total_price_tax_incl = (float) $orderDetail->total_price_tax_excl + $totalAmount;
```

Reply:
`$orderDetail->total_price_tax_excl` isn't 0. It has the good tax excl value. So it is ok to say that tax included total price = tax excluded total price + total tax amount.

But you're right, it is mostly unreadable, and I think there is another problem on products with multiples taxesAmount. Their `$orderDetail->unit_price_tax_incl ` will store the amount of the LAST taxeAmount, and not a sum of every taxeAmount.

Perhaphs the update can be done like this:
- parse all taxesAmount, to populate `order_detail_tax` table
- at the end, update `order_detail_tax` with the new calculated taxes (or `0` if there was no tax)

Here is a try:

```php
// FIX: new variable to store the sum of "unit amount" taxes
//$unitAmount = $totalAmount = 0;
$unitAmount = $totalUnitAmount = $totalAmount = 0;
// END
if (!empty($taxesAmount)) {
    $orderDetailTaxes = [];
    foreach ($taxesAmount as $taxId => $amount) {
        switch ($roundType) {
            case Order::ROUND_ITEM:
                $unitAmount = (float) Tools::ps_round($amount, $computingPrecision);
                $totalAmount = $unitAmount * $orderDetail->product_quantity;

                break;
            case Order::ROUND_LINE:
                $unitAmount = $amount;
                $totalAmount = Tools::ps_round($unitAmount * $orderDetail->product_quantity, $computingPrecision);

                break;
            case Order::ROUND_TOTAL:
                $unitAmount = $amount;
                $totalAmount = $unitAmount * $orderDetail->product_quantity;

                break;
        }
        $orderDetailTaxes[] = [
            'id_order_detail' => $orderDetail->id,
            'id_tax' => $taxId,
            'unit_amount' => (float) $unitAmount,
            'total_amount' => (float) $totalAmount,
        ];
        // FIX: store the current unit amount, and do not alter orderDetail for now
        $totalUnitAmount += $unitAmount;
        //$orderDetail->unit_price_tax_incl = (float) $orderDetail->unit_price_tax_excl + $unitAmount;
        //orderDetail->total_price_tax_incl = (float) $orderDetail->total_price_tax_excl + $totalAmount;
        // END
    }

    Db::getInstance()->insert('order_detail_tax', $orderDetailTaxes, false);
}

// Update OrderDetail values
// FIX: store the sum of each taxeAmount
//$orderDetail->unit_price_tax_incl = (float) $orderDetail->unit_price_tax_excl + $unitAmount;
$orderDetail->unit_price_tax_incl = (float) $orderDetail->unit_price_tax_excl + $totalUnitAmount;
// END
$orderDetail->total_price_tax_incl = (float) $orderDetail->total_price_tax_excl + $totalAmount;
if (!$orderDetail->update()) {
    throw new OrderException('An error occurred while editing the product line.');
}
```

Reply:
@Jeremie-Kiwik The name of the variable confused me. I would expect `totalTaxAmount`, but on the other hand method is named `updateOrderDetailTaxes`, so maybe I'm nitpicking

> and I think there is another problem on products with multiples taxesAmount. Their $orderDetail->unit_price_tax_incl  will store the amount of the LAST taxeAmount, and not a sum of every taxeAmount.

If I recall correctly, we have scenarios for this in behat tests, I'll ping @jolelievre and @amine-O because I believe that this error is a major one

Reply:
So, here is a try with readable names + will store in `order_detail` the sum of every taxeAmount. Which is a breaking change :-/